File: benchmarks/python/file_processing.py
#!/usr/bin/env python3
"""BENCH_FILE_LINES=5000"""
from pathlib import Path


def file_workload(path: Path, lines: int) -> int:
    body = "".join("benchmark-line\n" for _ in range(lines))
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(body, encoding="utf-8")
    return len(path.read_text(encoding="utf-8"))


def main() -> None:
    target = Path("benchmarks/results/tmp/bench-io.txt")
    print(file_workload(target, 5000))


if __name__ == "__main__":
    main()

File: benchmarks/rust/src/bin/file_processing.rs
use std::fs;
use std::path::PathBuf;

fn file_workload(path: &PathBuf, lines: i32) -> usize {
    let body: String = (0..lines).map(|_| "benchmark-line\n").collect();
    if let Some(parent) = path.parent() {
        let _ = fs::create_dir_all(parent);
    }
    fs::write(path, &body).expect("write");
    fs::read_to_string(path).expect("read").len()
}

fn main() {
    let path = PathBuf::from("benchmarks/results/tmp/bench-io.txt");
    println!("{}", file_workload(&path, 5000));
}
